lib/spack/spack/package.py

```diff
@@ -1063,7 +1064,7 @@ def do_activate(self, force=False):
 
         # Activate any package dependencies that are also extensions.
         if not force:
-            for spec in self.spec.traverse(root=False):
+            for spec in self.spec.traverse(root=False, deptype=spack.DEP_RUN):
```

Comment:
this dep type should match the `xxx(deptype) DEP_?` comment above -- it's either link, run, or both.  I think you want both, as some packages need libs from the extendee as well as things like python files.